File: app/src/main/java/jp/hazuki/yuzubrowser/webkit/handler/WebSrcImageOpenOtherAppHandler.java
package jp.hazuki.yuzubrowser.webkit.handler;

import android.content.Context;

import java.lang.ref.WeakReference;

import jp.hazuki.yuzubrowser.R;
import jp.hazuki.yuzubrowser.utils.PackageUtils;

public class WebSrcImageOpenOtherAppHandler extends WebSrcImageHandler {
    private final WeakReference<Context> mReference;

    public WebSrcImageOpenOtherAppHandler(Context activity) {
        mReference = new WeakReference<>(activity);
    }

    @Override
    public void handleUrl(String url) {
        Context context = mReference.get();
        if (context != null) {
            context.startActivity(PackageUtils.createChooser(context, url, context.getText(R.string.open_other_app)));
        }
    }
}

File: app/src/main/java/jp/hazuki/yuzubrowser/webkit/handler/WebSrcImageOpenOtherAppHandler.kt
package jp.hazuki.yuzubrowser.webkit.handler

import android.content.Context
import jp.hazuki.yuzubrowser.R
import jp.hazuki.yuzubrowser.utils.PackageUtils
import java.lang.ref.WeakReference

class WebSrcImageOpenOtherAppHandler(activity: Context) : WebSrcImageHandler() {
    private val mReference: WeakReference<Context> = WeakReference(activity)

    override fun handleUrl(url: String) {
        mReference.get()?.run {
            startActivity(PackageUtils.createChooser(this, url, getText(R.string.open_other_app)))
        }
    }
}
